test/jdk/javax/swing/JRadioButton/bug4380543.java

```diff
@@ -0,0 +1,233 @@
+    static final CountDownLatch latch = new CountDownLatch(1);
+    private static AtomicReference<Boolean> testResult = new AtomicReference<>(false);
+
+    public static void main(String args[]) throws Exception {
+
+        SwingUtilities.invokeAndWait(new Runnable() {
+            public void run() {
+                try {
+                    testObj = new testFrame();
+                    createUI();
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+
+        boolean status = latch.await(5, TimeUnit.MINUTES);
+
+        if (!status) {
+            System.out.println("Test timed out.");
+        }
+
+        onCompletion(testResult);
+
+    }
+
+    public static void createUI() throws Exception {
+        frame = new JFrame();
+        JPanel mainControlPanel = new JPanel(new BorderLayout());
+        JPanel resultButtonPanel = new JPanel(new FlowLayout());
+
+        JTextArea instructionTextArea = new JTextArea();
+
+        String instructions
+                = "INSTRUCTIONS:" +
+                "\n 1. This is a Windows specific test. If you are not on " +
+                "Windows, press Pass." +
+                "\n 2. Check if the Left insets(margins) is set visually " +
+                "similar to other three sides around Radio Button and CheckBox" +
+                "(insets set to 20 on all 4 sides)." +
+                "\n 3. If Left insets(margins) appear Empty, press Fail, " +
+                "else press Pass.";
+
+        instructionTextArea.setText(instructions);
+        instructionTextArea.setEnabled(false);
+        instructionTextArea.setDisabledTextColor(Color.black);
+        instructionTextArea.setBackground(Color.white);
+
+        mainControlPanel.add(instructionTextArea,BorderLayout.NORTH);
+        JButton passButton = new JButton("Pass");
+        passButton.setActionCommand("Pass");
+
+        passButton.addActionListener((ActionEvent e) -> {
+            testResult.set(true);
+            latch.countDown();
+        });
+
+        JButton failButton = new JButton("Fail");
+        failButton.setActionCommand("Fail");
+        failButton.addActionListener((ActionEvent e) -> {
+            testResult.set(false);
+            latch.countDown();
+        });
+
+        resultButtonPanel.add(passButton);
+        resultButtonPanel.add(failButton);
+        mainControlPanel.add(resultButtonPanel);
+        frame.getContentPane().add(mainControlPanel,BorderLayout.SOUTH);
+
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        frame.setLocation(500,50);
+        frame.setSize(400, 150);
+        frame.pack();
+        frame.setVisible(true);
+
+        Thread.sleep(1000);
+    }
+
+    private static void disposeUI()
+    {
+        testObj.dispose();
+        frame.dispose();
+    }
+    private static void onCompletion(AtomicReference<Boolean> res)
+    {
+        disposeUI();
+        if (res.toString() == "false")
```

Comment:
This look weird. Why can't you use `res.get()`? It returns a `Boolean` object which can be used directly in an if-condition.

Reply:
Updated as per the review comments @aivanov-jdk .